### PR TITLE
feat: assign names to repos

### LINKED ISSUE #1388: Assign names to repos
Repo URLs can be hard to read. Even the ones BorgBase uses. So would suggest to ask for a name when adding a repo and show that in the dropdown and other places.

### PR DESCRIPTION
**Pull request by diivi**
### Description
Adds a "name" field in the new repo creation form and displays it where I thought it was suitable.

### Related Issue
Fixes #1388 

### Motivation and Context
Easier to remember and tag repos.

### How Has This Been Tested?
Tried creating a repo and looking at the places where I rendered the name instead of the url.

### Screenshots (if appropriate):

image
image
image

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] I have read the [CONTRIBUTING](https://vorta.borgbase.com/contributing/) guide.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

*I provide my contribution under the terms of the [license](./../../LICENSE.txt) of this repository and I affirm the [Developer Certificate of Origin][dco].*

[dco]: https://developercertificate.org/

### CODE REVIEW

**Review by real-yfprojects — COMMENTED** [review]
I think the repo id should be used when the user didn't specify a repo name.

**Comment by m3nu** [comment]
I'd still show the repo URL **behind** the name in the dropdown and on the archives tab.

And have "optional" in the placeholder. E.g. `Some name (optional)`. To clarify 1) that this is optional and 2) that this is used only locally to manage the repo. Maybe there is better wording for it.

**Comment by real-yfprojects** [comment]
I think a tooltip showing the complete repo path would be useful since it sometimes doesn't fit into the displaying widget.

**Comment by real-yfprojects** [comment]
Looking good. However I am unsure whether I like using the repo id as a repo name for existing repos. I don't see any value in having 1,2,3 and so on in front of my repo paths/urls. I believe its better to only show the name if the user specified one.

And we should provide a way to edit the repo name. This could be a task for a future PR.

File: src/vorta/views/archive_tab.py [hunk 1]
```diff
@@ -251,7 +251,7 @@ def populate_from_profile(self):
             if repo_mount_points:
                 self.repo_mount_point = repo_mount_points[0]
 
-            self.toolBox.setItemText(0, self.tr('Archives for %s') % profile.repo.url)
+            self.toolBox.setItemText(0, self.tr('Archives for {}').format(f"{profile.repo.name} ({profile.repo.url})"))
```
Comment (real-yfprojects) [inline src/vorta/views/archive_tab.py:254]: Double space in case of empty profile name.

**Review by real-yfprojects — DISMISSED** [review]

**Comment by m3nu** [comment]
Implementation looks good and it does what it promises.

Just have one idea to make it even better: How about adding some space or a thin line before the 2 password fields? To slightly separate them from the other fields and have a logical grouping.

Screenshot 2023-05-29 at 11 06 54

**Comment by diivi** [comment]
With separators (https://github.com/borgbase/vorta/pull/1665/commits/2e27718e13f96975d42ba58e262be465bc8f8788)

image
image

**Comment by m3nu** [comment]
I think it's great like this.

**Review by m3nu — DISMISSED** [review]

**Comment by real-yfprojects** [comment]
I propose this:

grafik

**Comment by m3nu** [comment]
This is OK to merge for me, but has pending feedback from @real-yfprojects regarding the small UI thing.

**Comment by diivi** [comment]
> I propose this, in accordance to the [HIG](https://develop.kde.org/hig/layout/metrics/):

Can you share the code/property?

**Comment by diivi** [comment]
image
Ignore the black eye icon, it seems to be an issue with the master branch as well, I will file it.

**Comment by m3nu** [comment]
Right. I had to update some icons due to licensing and forgot to adjust the `fill` property for those. Please adjust it by adding `fill="#000000"` in the first `<svg>` tag:

`<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 1536 1536" style="enable-background:new 0 0 1536 1536;" fill="#000000" xml:space="preserve">`

This is needed for all icons, so they can be colored light/dark.

**Comment by diivi** [comment]
Done

**Comment by m3nu** [comment]
Thanks for sneaking this in. Will test locally in a few hours and then merge.

**Review by m3nu — APPROVED** [review]

**Comment by m3nu** [comment]
Works as expected locally and implementation looks reasonable.

Congrats on another successful PR, @diivi 👏

**Pull request merged**